src/gui/foldercreationdialog.cpp

```diff
@@ -37,7 +37,7 @@ FolderCreationDialog::FolderCreationDialog(const QString &destination, QWidget *
 
     const QString suggestedFolderNamePrefix = QObject::tr("New folder");
 
-    const auto newFolderFullPath = _destination + "/" + suggestedFolderNamePrefix;
+    const QString newFolderFullPath = _destination + "/" + suggestedFolderNamePrefix;
```

Comment:
@FlexW I am a bit confused about this because `newFolderFullPath` is then just passed to `QDir(const QString& path)` that should result in a similar implicit conversion from `QStringBuilder` to `QString`. But, if this works, let's move it forward.

Maybe @er-vin has some idea why this is happening?

Reply:
The problem is that QStringBuilder is lazy evaluated and that the temporaries used in the expression (`_destination`, `"/"` and `suggestedFolderNamePrefix`) are long gone when the line of the `QDir` ctor is reached and the conversion to QString occurs.

BTW, it was here before but it should be `QLatin1Char('/')` instead of `"/"`.

Reply:
> BTW, it was here before but it should be QLatin1Char('/') instead of "/".

@er-vin @FlexW I agree.